### PR TITLE
Fix refund transfer for issue 1429

### LINKED ISSUE #1429: Fix the last failing tests for the Matrix PR
## Problem Definition

Two tests were failing in the [matrix PR](https://github.com/raiden-network/raiden/pull/1373). 

- integration/transfer/test_refund_transfer.py::test_refund_messages
- integration/transfer/test_refund_transfer.py::test_refund_transfer

Fix them

### PR DESCRIPTION
**Pull request by LefterisJP**
Fix #1429 

- Also clean up the `state.payment_mapping.secrethashes_to_task` dict properly for end states so that assertions are not hit when we get subsequent inputs in the state machine.

- Also increase `network_wait` for tests that run on Travis since one test failed due to timeout there while all events appeared to have been processed fine. Log [here](https://gist.github.com/LefterisJP/9349fdc10be00c0f195ecb2222c561f7). Travis is really slow/unstable.

### CODE REVIEW

File: raiden/transfer/node.py [hunk 1]
```diff
@@ -171,6 +172,9 @@ def subdispatch_to_paymenttask(node_state, state_change, secrethash):
                 )
                 events = sub_iteration.events
 
+        if sub_iteration and sub_iteration.new_state is None:
+            del node_state.payment_mapping.secrethashes_to_task[secrethash]
```
Comment (palango) [inline raiden/transfer/node.py]: I'm having problems to understand why this is needed. Does `sub_iteration.new_state == None` signal that the payment is finished and we subsequently clean up the task from the `secrethashes_to_task` mapping? Is there more stuff that should be removed in this case?

Comment (hackaugusto) [reply raiden/transfer/node.py]: > Does sub_iteration.new_state == None signal that the payment is finished and we subsequently clean up the task from the secrethashes_to_task mapping?

Yes, the task finished, either successfully or not, so we need to clean up after it.]

> Is there more stuff that should be removed in this case?

Not that i recall.

Comment (hackaugusto) [reply raiden/transfer/node.py]: @LefterisJP could you also do the cleanup for the functions

- subdispatch_initiatortask
- subdispatch_mediatortask
- subdispatch_targettask

Comment (palango) [reply raiden/transfer/node.py]: > Yes, the task finished, either successfully or not, so we need to clean up after it.]

Are possible errors handled somewhere else?

**Comment by LefterisJP** [comment]
> could you also do the cleanup for the functions

Yep! Thanks Augusto, this is why I wanted you to check the PR, to see where else I missed it.

File: raiden/transfer/node.py (hunk 1 above)
Comment (LefterisJP) [reply raiden/transfer/node.py]: @palango Paul this is to avoid this assertion: https://gist.github.com/LefterisJP/9349fdc10be00c0f195ecb2222c561f7

**Pull request merged**